### ISSUE TITLE
Can't run doctest with GHC 8.4.1/8.4.2 on Mac OSX

### ISSUE THREAD

**Issue by LeventErkok** [sol/doctest#199]
I'm pretty sure this has everything to do with GHC 8.4.1 on Mac instead of doctest itself, but I'm having difficulty running doctest after upgrading to GHC 8.4.1:

```
$ doctest Data/SBV/Examples/Polynomials/Polynomials.hs
doctest:
lookupSymbol failed in relocateSection (RELOC_GOT)
/usr/local/lib/ghc-8.4.1/integer-gmp-1.0.1.0/HSinteger-gmp-1.0.1.0.o: unknown symbol `___gmp_rands'
doctest: doctest: unable to load package `integer-gmp-1.0.1.0'
```

The following ticket (observed in haddock) seems to be the same issue, but for `haddock`: https://github.com/haskell/haddock/issues/767

I'm curious if you tried doctest on Mac with GHC 8.4.1 and have run into this. Are there any known workarounds?

**Comment by sol** [comment]
Hey @LeventErkok thank you for taking the time to open this issue!

> I'm curious if you tried doctest on Mac with GHC 8.4.1 and have run into this.

I don't have access to Mac hardware, so I can't help with this.  What we could do is add a OS X build to travis to detect this kind of issues earlier in the future.

**Comment by LeventErkok** [comment]
I'm afraid the problem is still there with ghc-8.4.2:

```
doctest:
lookupSymbol failed in relocateSection (RELOC_GOT)
/usr/local/lib/ghc-8.4.2/integer-gmp-1.0.2.0/HSinteger-gmp-1.0.2.0.o: unknown symbol `___gmp_rands'
doctest: doctest: unable to load package `integer-gmp-1.0.2.0'
```

Can anyone else confirm?

**Comment by LeventErkok** [comment]
Sadly this is still an issue with GHC 8.4.2. Progress seems to have stalled on it.

**Comment by quasicomputational** [comment]
The GHC issue has been resolved so I'm going to assume this is no longer a problem.

**Comment by TomMD** [comment]
That's an interesting assumption. Which version of GHC claims to have fixed the issue?  I can produce the same failure in another project with OS-X and GHC 8.10.1

**Comment by quasicomputational** [comment]
The GHC bug suggests maybe 8.6.3 had the fix; IIRC 8.10.1 had some fun bugs of its own - is it still reproducing with 8.10.3?

**Comment by TomMD** [comment]
I saw 8.10.3 has a linker issue fixed but haven't updated the project from 8.10.1.  When I next move the compiler, probably to 9.0, I can find out and report back.
